src/ray/core_worker/core_worker.cc

```diff
@@ -93,24 +100,21 @@ CoreWorker::CoreWorker(const WorkerType worker_type, const Language language,
   profiler_ = std::make_shared<worker::Profiler>(worker_context_, node_ip_address,
                                                  io_service_, gcs_client_);
 
-  // Initialize task execution.
+  // Initialize task receivers.
+  auto execute_task = std::bind(&CoreWorker::ExecuteTask, this, std::placeholders::_1,
+                                std::placeholders::_2, std::placeholders::_3);
   if (worker_type_ == WorkerType::WORKER) {
     RAY_CHECK(task_execution_callback_ != nullptr);
-
-    // Initialize task receivers.
-    auto execute_task = std::bind(&CoreWorker::ExecuteTask, this, std::placeholders::_1,
-                                  std::placeholders::_2, std::placeholders::_3);
     raylet_task_receiver_ = std::unique_ptr<CoreWorkerRayletTaskReceiver>(
         new CoreWorkerRayletTaskReceiver(raylet_client_, execute_task, exit_handler));
     direct_actor_task_receiver_ = std::unique_ptr<CoreWorkerDirectActorTaskReceiver>(
         new CoreWorkerDirectActorTaskReceiver(worker_context_, task_execution_service_,
-                                              worker_server_, execute_task,
-                                              exit_handler));
-    worker_server_.RegisterService(grpc_service_);
+                                              execute_task, exit_handler));
   }
+  core_worker_server_.RegisterService(grpc_service_);
```

Comment:
should be moved to the if block above as driver doesn't need to register service?

Reply:
if it should not, please add a comment on why?

Reply:
The driver still needs to receive RPC responses to its worker requests, and in the future we will add other RPCs such as statistics collection. I don't think this needs a comment moving forward it's the "expected setup".

Reply:
do we have any guards against sending non-desired requests to a driver? e.g. a PushTaskRequest.  since a driver now has a RPC server.